pandas/core/window/rolling.py

```diff
@@ -247,8 +247,33 @@ def __repr__(self) -> str:
         return f"{self._window_type} [{attrs}]"
 
     def __iter__(self):
-        url = "https://github.com/pandas-dev/pandas/issues/11704"
-        raise NotImplementedError(f"See issue #11704 {url}")
+        window = self._get_window(win_type=None)
+
+        blocks, obj = self._create_blocks()
+        block_list = list(blocks)
+        index = self._get_window_indexer(window=window)
+
+        # Choose the min between min_periods and window to determine the output size
+        if self.min_periods is None:
+            iter_threshold = window
+        else:
+            iter_threshold = min(window, self.min_periods)
+
+        for block in block_list:
```

Comment:
I think it'd be more intuitive for the user if the window was over the entire object instead of per block.

Reply:
thanks, changed to `obj`!